internal/provider/namespace_resource.go

```diff
@@ -278,18 +283,34 @@ func (r *namespaceResource) Create(ctx context.Context, req resource.CreateReque
 			return
 		}
 	}
+
+	var spec = &namespacev1.NamespaceSpec{
+		Name:          plan.Name.ValueString(),
+		Regions:       regions,
+		RetentionDays: int32(plan.RetentionDays.ValueInt64()),
+		CodecServer:   codecServer,
+	}
+
+	if plan.ApiKeyAuth.ValueBool() {
+		spec.ApiKeyAuth = &namespacev1.ApiKeyAuthSpec{Enabled: true}
+	} else {
+		if plan.AcceptedClientCA.IsNull() {
+			resp.Diagnostics.AddError("accepted_client_ca is required when API key authentication is disabled", "")
```

Comment:
nit on error message - Namespace not configured with authentication.  accepted_client_ca is required when API key authentication is not enabled (api_key_auth is not set to true).